tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -298,3 +446,80 @@ def test_to_hierarchical_dataset_hierarchical_structure(
         expected_level_segments = hierarchical_structure_complex.get_level_segments(level_name=level_name)
         obtained_level_segments = hs.get_level_segments(level_name=level_name)
         assert sorted(obtained_level_segments) == sorted(expected_level_segments)
+
+
+@pytest.mark.parametrize(
```

Comment:
Change the current fixtures or create separate with
- NaNs in the beginning of some segments
- NaNs in the middle of some segments
We should make sure that everything works with NaNs